app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,24 @@
+class CommentsController < ApplicationController
+  before_action :logged_in_user, only: [:create]
+
+  def create
+    post = Post.find(params[:comment][:post_id])
+    if params[:comment][:comment_id].present?
+      parent = Comment.find(params[:comment][:comment_id])
+    else
+      parent = post
+    end
+    comment = parent.comments.build
+    if (comment.update_attributes(comment_params))
+      comment.save
```

Comment:
this is looking a lot better, I'd probably recommend extracting out the parent assignment into it's own method so the content of the `create` action stays more focused.

would also make `comment_params` (and potentially a `parent`/`commentable` method you add) private methods. Convention states that public methods in controllers should line up with available application actions